com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/handler/LaunchRequestHandler.java

```diff
@@ -91,11 +94,30 @@ public void handle(Command command, Arguments arguments, Response response, IDeb
         }
         sourceProvider.initialize(options);
 
+        // Append environment to native environment.
+        String[] envp = null;
+        if (launchArguments.env != null && !launchArguments.env.isEmpty()) {
+            Map<String, String> environment = new HashMap<>(System.getenv());
+            for (Entry<String, String> entry : launchArguments.env.entrySet()) {
+                // For duplicated variable, show a warning message.
+                if (environment.containsKey(entry.getKey())) {
+                    logger.warning(String.format("There is the same variable '%s' existed in the system environment, "
+                            + "the program will use the newly configured variable to override the system environment variable.", entry.getKey()));
```

Comment:
There are duplicated environment variables. The values specified in launch.json will be used. Here are the duplicated entries: env1, env2.